### PR TITLE
Extract `arel_visitor` and move up to the abstract adapter

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract_adapter.rb [hunk 1]
```diff
@@ -143,6 +149,10 @@ def collector
         end
       end
 
+      def arel_visitor # :nodoc:
+        (Arel::Visitors::VISITORS[@config[:adapter]] || Arel::Visitors::ToSql).new(self)
```
Comment (yui-knk) [inline activerecord/lib/active_record/connection_adapters/abstract_adapter.rb]: Why fallback to `Arel::Visitors::ToSql`?
I think it is better to override this method if someone create an adapter that Arel does not support.

Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/abstract_adapter.rb]: This code is derived from https://github.com/rails/arel/blob/v7.0.0/lib/arel/visitors.rb#L33.

Comment (yui-knk) [inline activerecord/lib/active_record/connection_adapters/abstract_adapter.rb]: I understand, thanks !

**Comment by jonatack** [comment]
Is this a permanent change? It breaks Ransack's Active Record adapter: `uninitialized constant Arel::Visitors::VISITORS`. Thanks :)

**Comment by kamipo** [comment]
hmm... `Arel::Visitors::VISITORS` was removed on arel master by https://github.com/rails/arel/pull/412...

**Comment by kamipo** [comment]
Fixed by #24634.

**Pull request merged**